source/opt/const_folding_rules.cpp

```diff
@@ -1545,6 +1545,66 @@ const analysis::Constant* FoldMax(const analysis::Type* result_type,
   return nullptr;
 }
 
+const analysis::Constant* FoldNMin(const analysis::Type* result_type,
+                                   const analysis::Constant* a,
+                                   const analysis::Constant* b,
+                                   analysis::ConstantManager*) {
+  if (const analysis::Float* float_type = result_type->AsFloat()) {
+    if (float_type->width() == 32) {
+      float va = a->GetFloat();
+      float vb = b->GetFloat();
+      if (std::isnan(va)) {
+        return b;
+      }
+      if (std::isnan(vb)) {
+        return a;
+      }
+      return (va < vb ? a : b);
+    } else if (float_type->width() == 64) {
+      double va = a->GetDouble();
+      double vb = b->GetDouble();
+      if (std::isnan(va)) {
+        return b;
+      }
+      if (std::isnan(vb)) {
+        return a;
+      }
+      return (va < vb ? a : b);
```

Comment:
Does this work for `-0` and `0`?

See the nmin documentation: https://registry.khronos.org/SPIR-V/specs/unified1/GLSL.std.450.html.

>  -0 compares less than +0.

I don't know if that is true in C++. In any case, add a test for it.

Reply:
I was originally basing the expectation of it working for `FMin`, which has the same spec.
Will add a test for that too.